### PR TITLE
MGMT-11712: collect  SOS reports from cluster nodes

### PR DESCRIPTION
**Pull request by eifrach**
* collecting SOS report from all nodes
* saving reports as artifact

### CODE REVIEW

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-compact-day2-masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.8-e2e-ai-operator-ztp-3masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-release-ocm-2.8-e2e-ai-operator-ztp-3masters

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
checking

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 1]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+      export VIRT_NET=$(virsh net-list --name  | grep -v default | tr -d "\n")
+      export TMP_HOSTFILE="/tmp/tmp_hostfile.json"
+
+      echo [] > ${TMP_HOSTFILE}
+      virsh net-dhcp-leases --network "${VIRT_NET}" | grep 'master\|worker' | awk '{ printf("%s/%s\n", $6, $5) }' | while read -r line
+      do
+          _HOST=$( echo "$line" | cut -d"/" -f1 )
+          _IP=$( echo "$line" | cut -d"/" -f2 )
+    
+          jq """. += [{ \"host\" : \"$_HOST\", \"address\" : \"$_IP\" }]""" < $TMP_HOSTFILE > $HOSTFILE
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:47]:
```suggestion
          jq """. += [{ \"host\" : \"${_HOST}\", \"address\" : \"${_IP}\" }]""" < ${TMP_HOSTFILE} > ${HOSTFILE}
```
nit: to be consistent with the rest of the file

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 2]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+    if [ $(virsh list --name |  tr -s '\n'  | wc -l) > 0 ]
+    then
+      export SOS_BASEDIR="/tmp/artifacts/sos"
+      export HOSTFILE="${SOS_BASEDIR}/virtual_hosts.json"
+      mkdir -p "${SOS_BASEDIR}"
+    
+      export VIRT_NET=$(virsh net-list --name  | grep -v default | tr -d "\n")
+      export TMP_HOSTFILE="/tmp/tmp_hostfile.json"
+
+      echo [] > ${TMP_HOSTFILE}
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:41]:
```suggestion
      echo "[]" > ${TMP_HOSTFILE}
```
I think it's clearer and not some shell shenanigans

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 3]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+      virsh net-dhcp-leases --network "${VIRT_NET}" | grep 'master\|worker' | awk '{ printf("%s/%s\n", $6, $5) }' | while read -r line
+      do
+          _HOST=$( echo "$line" | cut -d"/" -f1 )
+          _IP=$( echo "$line" | cut -d"/" -f2 )
+    
+          jq """. += [{ \"host\" : \"$_HOST\", \"address\" : \"$_IP\" }]""" < $TMP_HOSTFILE > $HOSTFILE
+          cat "$HOSTFILE" > "$TMP_HOSTFILE"
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:48]:
```suggestion
          cat "${HOSTFILE}" > "${TMP_HOSTFILE}"
```

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 4]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+      done
+    
+      jq -c ".[]" ${HOSTFILE} | while read ITEM 
+      do
+        _HOST=$(echo $ITEM | jq -r .host )
+        _IP=$(echo $ITEM | jq -r .address )
+
+        export REPORT_PATH="${SOS_BASEDIR}/$_HOST"
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:56]:
```suggestion
        _HOST=$(echo ${ITEM} | jq -r .host )
        _IP=$(echo ${ITEM} | jq -r .address )

        export REPORT_PATH="${SOS_BASEDIR}/${_HOST}"
```

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 5]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+        mkdir -p ${REPORT_PATH}; chmod o+w ${REPORT_PATH} 
+
+        timeout -s 9 30m ssh -o StrictHostKeyChecking=off -l core $_IP REPORT_PATH="${SOS_BASEDIR}/$_HOST" bash - << "EOF"
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:59]:
```suggestion
        timeout -s 9 30m ssh -o StrictHostKeyChecking=off -l core $_IP REPORT_PATH="${SOS_BASEDIR}/${_HOST}" bash - << "EOF"
```

**Review by adriengentil — CHANGES_REQUESTED** [review]
some nits, can you also add the jira ticket to the title of the PR?

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

**Comment by eifrach** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-ai-operator-ztp

File: ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh [hunk 6]
```diff
@@ -25,6 +26,51 @@ function getlogs() {
...
+        export REPORT_PATH="${SOS_BASEDIR}/${_HOST}"
+        mkdir -p ${REPORT_PATH}; chmod o+w ${REPORT_PATH} 
+
+        timeout -s 9 30m ssh -o StrictHostKeyChecking=off -l core $_IP REPORT_PATH="${SOS_BASEDIR}/${_HOST}" bash - << "EOF"
+          mkdir -p ${REPORT_PATH} 
+    
+          yes | toolbox "sos report --batch --tmp-dir ${REPORT_PATH} -k crio.all=on -k crio.logs=on -k openshift.host=on -k openshift.podlogs=on \
+              -o  openshift,openshift_ovn,crio,containers_common,host,containerd,logs"
+
+            sudo chmod -R +r "${REPORT_PATH}"
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/baremetal/operator/gather/assisted-baremetal-operator-gather-commands.sh:65]:
```suggestion
           sudo chmod -R +r "${REPORT_PATH}"
```
nit, looks like it's not aligned

**Review by adriengentil — COMMENTED** [review]
LGTM, just a nit

**Review by adriengentil — APPROVED** [review]

**Pull request merged**